From 9142ec5d05d2a4702d45ab38a1d37be9be13059a Mon Sep 17 00:00:00 2001
From: Chetan Rao <chetan.rao@intel.com>
Date: Mon, 15 Jul 2013 15:27:57 +0300
Subject: dalvik: Search for x86-atom and x86-slm in TARGET_ARCH_VARIANT

BZ: 120393

TARGET_ARCH_VARIANT now supports x86-slm and haswell. Need to search for x86-atom
and x-86-slm in Android.mk

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I75f5a4f97210e7b1161257751a5d7477b593f617
Orig-MCG-Change-Id: I10ca5ccafa02942ec6b17b20d64aed4fdac5f862
Signed-off-by: Chetan Rao <chetan.rao@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/Dvm.mk            |   14 +++++++++++---
 vm/ReconfigureDvm.mk |    4 ++--
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/vm/Dvm.mk b/vm/Dvm.mk
index 1be4a7a..4ef96dc 100644
--- a/vm/Dvm.mk
+++ b/vm/Dvm.mk
@@ -47,8 +47,16 @@ ifneq ($(strip $(DEBUG_DALVIK_VM)),)
   dvm_make_debug_vm := $(DEBUG_DALVIK_VM)
 endif
 
-# Disable x86-atom variant
-ifeq ($(dvm_arch_variant),x86-atom)
+# Disable x86-atom/x86-slm/haswell variant
+ifneq (, $(findstring x86-, $(dvm_arch_variant)))
+dvm_arch_variant := x86
+endif
+
+ifneq (, $(findstring haswell, $(dvm_arch_variant)))
+dvm_arch_variant := x86
+endif
+
+ifneq (, $(findstring silvermont, $(dvm_arch_variant)))
 dvm_arch_variant := x86
 endif
 
@@ -265,7 +273,7 @@ LOCAL_C_INCLUDES += \
 	dalvik \
 	dalvik/vm \
 	external/zlib \
-	libcore/include \
+	libcore/include
 
 MTERP_ARCH_KNOWN := false
 
diff --git a/vm/ReconfigureDvm.mk b/vm/ReconfigureDvm.mk
index 2ac8f54..1b4f58c 100644
--- a/vm/ReconfigureDvm.mk
+++ b/vm/ReconfigureDvm.mk
@@ -19,8 +19,8 @@ dvm_os := $(TARGET_OS)
 dvm_arch := $(TARGET_ARCH)
 dvm_arch_variant := $(TARGET_ARCH_VARIANT)
 
-# Disable x86-atom variant
-ifeq ($(dvm_arch_variant),x86-atom)
+# Disable x86-atom or x86-slm variant
+ifneq (, $(findstring x86-, $(dvm_arch_variant)))
 dvm_arch_variant := x86
 endif
 
-- 
1.7.4.1

